plugins/node/opentelemetry-instrumentation-undici/src/undici.ts

```diff
@@ -0,0 +1,460 @@
+    // ours is called. So we need to ensure the property is initalized
+    this._channelSubs = this._channelSubs || [];
+    this.subscribeToChannel(
+      'undici:request:create',
+      this.onRequestCreated.bind(this)
+    );
+    this.subscribeToChannel(
+      'undici:client:sendHeaders',
+      this.onRequestHeaders.bind(this)
+    );
+    this.subscribeToChannel(
+      'undici:request:headers',
+      this.onResponseHeaders.bind(this)
+    );
+    this.subscribeToChannel('undici:request:trailers', this.onDone.bind(this));
+    this.subscribeToChannel('undici:request:error', this.onError.bind(this));
+  }
+
+  override setConfig(config?: UndiciInstrumentationConfig): void {
+    super.setConfig(config);
+
+    if (config?.enabled) {
+      this.enable();
+    } else {
+      this.disable();
+    }
+  }
+
+  protected override _updateMetricInstruments() {
+    this._httpClientDurationHistogram = this.meter.createHistogram(
+      'http.client.request.duration',
+      {
+        description: 'Measures the duration of outbound HTTP requests.',
+        unit: 'ms',
+        valueType: ValueType.DOUBLE,
+      }
+    );
+  }
+
+  private _getConfig(): UndiciInstrumentationConfig {
+    return this._config as UndiciInstrumentationConfig;
+  }
+
+  private subscribeToChannel(
+    diagnosticChannel: string,
+    onMessage: ListenerRecord['onMessage']
+  ) {
+    const channel = diagch.channel(diagnosticChannel);
+    channel.subscribe(onMessage);
+    this._channelSubs.push({
+      name: diagnosticChannel,
+      channel,
+      onMessage,
+    });
+  }
+
+  // This is the 1st message we receive for each request (fired after request creation). Here we will
+  // create the span and populate some atttributes, then link the span to the request for further
+  // span processing
+  private onRequestCreated({ request }: RequestMessage): void {
+    // Ignore if:
+    // - instrumentation is disabled
+    // - ignored by config
+    // - method is 'CONNECT'
+    const config = this._getConfig();
+    const shouldIgnoreReq = safeExecuteInTheMiddle(
+      () =>
+        !config.enabled ||
+        request.method === 'CONNECT' ||
+        config.ignoreRequestHook?.(request),
+      e => e && this._diag.error('caught ignoreRequestHook error: ', e),
+      true
+    );
+
+    if (shouldIgnoreReq) {
+      return;
+    }
+
+    const startTime = hrTime();
+    const rawHeaders = request.headers.split('\r\n');
+    const reqHeaders = new Map(
+      rawHeaders.map(h => {
+        const sepIndex = h.indexOf(':');
+        const name = h.substring(0, sepIndex).toLowerCase();
+        const val = h.substring(sepIndex + 1).trim();
+        return [name, val];
+      })
+    );
+
+    const requestUrl = new URL(request.origin + request.path);
+    const urlScheme = requestUrl.protocol.replace(':', '');
+    const attributes: Attributes = {
+      [SemanticAttributes.HTTP_REQUEST_METHOD]: request.method,
```

Comment:
Note the discussion of normalizing the `http.request.method` to the known upper-case values, and falling back to `_OTHER` -- and then if normalizing, then setting `http.request.method_original`. 
https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/

On *can* create an HTTP request with undici with a method other than the "known" set:

```js
    const res = await undici.request(
        'http://localhost:8200/',
        {
            method: 'WALLA', // or 'GeT'
            headers: {
                'Spam': 'Eggs'
            }
        });
```

Reply:
done, thanks for spotting it